src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let stringFromThis = '';
+    const pseudoSeparator = separator;
+
+    for (const n of this) {
+      let item = n;
+
+      if (n === undefined || n === null) {
+        item = '';
+      }
+
+      stringFromThis += item;
+
+      if (this.indexOf(n) !== this.length - 1) {
+        stringFromThis += pseudoSeparator;
+      }
```

Comment:
Why to search for index of element (and iterate your array to get it) if you can write `for  in` loop and always have index?